From 1222d345b16541bf282b8df6d9ccbe8b3fbfbafc Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Thu, 15 Jan 2015 13:42:16 +0800
Subject: [PATCH] config: board: amend the "Bus Speed" choices dependence

Add the CPU clock as one of dependence to match the Bus Speed with the CPU Clock.

Signed-off-by: Wenyou Yang <wenyou.yang@atmel.com>
---
 board/Config.in | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/board/Config.in b/board/Config.in
index 6852863..ca95ff4 100644
--- a/board/Config.in
+++ b/board/Config.in
@@ -278,35 +278,35 @@ choice
 
 config CONFIG_BUS_SPEED_83MHZ
 	bool "83 MHz"
-	depends on SUPPORT_BUS_SPEED_83MHZ
+	depends on SUPPORT_BUS_SPEED_83MHZ && CONFIG_CPU_CLK_166MHZ
 
 config CONFIG_BUS_SPEED_90MHZ
 	bool "90 MHz"
-	depends on SUPPORT_BUS_SPEED_90MHZ
+	depends on SUPPORT_BUS_SPEED_90MHZ && CONFIG_CPU_CLK_180MHZ
 
 config CONFIG_BUS_SPEED_100MHZ
 	bool "100 MHz"
-	depends on SUPPORT_BUS_SPEED_100MHZ
+	depends on SUPPORT_BUS_SPEED_100MHZ && (CONFIG_CPU_CLK_200MHZ || CONFIG_CPU_CLK_400MHZ)
 
 config CONFIG_BUS_SPEED_133MHZ
 	bool "133 MHz"
-	depends on SUPPORT_BUS_SPEED_133MHZ
+	depends on SUPPORT_BUS_SPEED_133MHZ && (CONFIG_CPU_CLK_528MHZ || CONFIG_CPU_CLK_400MHZ || CONFIG_CPU_CLK_396MHZ || CONFIG_CPU_CLK_266MHZ)
 
 config CONFIG_BUS_SPEED_148MHZ
 	bool "148 MHz"
-	depends on SUPPORT_BUS_SPEED_148MHZ
+	depends on SUPPORT_BUS_SPEED_148MHZ && CONFIG_CPU_CLK_444MHZ
 
 config CONFIG_BUS_SPEED_166MHZ
 	bool "166 MHz"
-	depends on SUPPORT_BUS_SPEED_166MHZ
+	depends on SUPPORT_BUS_SPEED_166MHZ && (CONFIG_CPU_CLK_498MHZ || CONFIG_CPU_CLK_332MHZ)
 
 config CONFIG_BUS_SPEED_170MHZ
 	bool "170 MHz"
-	depends on SUPPORT_BUS_SPEED_170MHZ
+	depends on SUPPORT_BUS_SPEED_170MHZ && CONFIG_CPU_CLK_510MHZ
 
 config CONFIG_BUS_SPEED_176MHZ
 	bool "176 MHz"
-	depends on SUPPORT_BUS_SPEED_176MHZ
+	depends on SUPPORT_BUS_SPEED_176MHZ && CONFIG_CPU_CLK_528MHZ
 
 endchoice
 
-- 
1.9.1

